### PR TITLE
Changes for new temple model

### PR DESCRIPTION
**Pull request by hdanak**
@somaholiday Looks like I don't have commit permissions on the new repo so I'm opening a pull request for now.

### CODE REVIEW

File: src/main/java/org/yomigae/model/FixtureCollection.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Collections;
+
+import com.google.common.collect.ForwardingList;
+import heronarts.lx.model.LXPoint;
+import heronarts.lx.model.LXFixture;
+
+
+public class FixtureCollection<F extends LXFixture> extends ForwardingList<F> implements LXFixture {
+	private List<F> fixtures = new ArrayList<>();
+	private final List<F> fixturesUnmodifiable = Collections.unmodifiableList(fixtures);
```
Comment (somaholiday) [inline src/main/java/org/yomigae/model/FixtureCollection.java:14]: What's the motivation for this unmodifiable list?

File: src/main/java/org/yomigae/YomigaeLayout.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+	// width of "wall washer" fixture, measured from first to last LED center
+	private static final double WALL_WASHER_WIDTH_FEET = 3; // TODO: use actual width
+
+	private final TempleBuilder builder = new TempleBuilder();
+
+	public YomigaeLayout() {
+		builder.defineFixtureType(FIXTURE_TYPE_SPOTLIGHT)
+			.addPoint(0, 0, 0)
+			.withDirection(0, 1, 0)
```
Comment (somaholiday) [inline src/main/java/org/yomigae/YomigaeLayout.java:40]: In practice this direction and the direction of the wall washers will be slightly more aimed toward the toriis, right?

File: src/main/java/org/yomigae/model/TempleModel.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+/**
+ * Class to encapsulate temple structure and points.
+ *
+ * Glossary:
+ *  - torii: gate structure comprised of two columns and a beam.
+ *  - torii types: T1 through T6, indicate the relative height of the torii
+ *  - column: vertical components of a torii
+ *  - column opening-width: head-on inside length between columns of a torii
+ *  - column width: head-on width of a torii's column
```
Comment (somaholiday) [inline src/main/java/org/yomigae/model/TempleModel.java:24]: Just to be clear, "head-on" in this case means "looking through the tunnel and hall (along the 12/6 o'clock axis)", while "side-view" means  "looking along the 3/9 o'clock axis", correct?

And on the XYZ orientation: it looks like you're using X as the 12/6 axis and Z as the 3/9?

**Review by somaholiday — APPROVED** [review]
Very cool.  Thanks for putting this together--really helps me understand about how to build knowledge into the LX model.

File: src/main/java/org/yomigae/model/FixtureCollection.java (hunk 1 above)
Comment (hdanak) [reply src/main/java/org/yomigae/model/FixtureCollection.java:14]: It's mainly to allow returning the list without the consumer modifying it (this way we don't have to make a copy every time).

File: src/main/java/org/yomigae/YomigaeLayout.java (hunk 2 above)
Comment (hdanak) [reply src/main/java/org/yomigae/YomigaeLayout.java:40]: Yeah I had a bunch of math in there to calculate the rotations but it was getting hairy. Later I realized that Apache Commons has some 3d rotation helpers that we can use, so I'll look into that later to make it more precise.

File: src/main/java/org/yomigae/model/TempleModel.java (hunk 3 above)
Comment (hdanak) [reply src/main/java/org/yomigae/model/TempleModel.java:24]: Yes I had it rotated incorrectly in my head when I first started writing this class. The comment should be updated, I'll do this when I also change the other variable names probably.

I believe what I had in mind by "head-on" was if you are looking at a torii as if you are to walk into the gate.

X is 6->12, Y is up, and Z is 3->9; this should also be documented.

**Pull request merged**